django/thunderstore/repository/models/comment.py

```diff
@@ -0,0 +1,44 @@
+from django.conf import settings
+from django.contrib.contenttypes.fields import GenericForeignKey
+from django.contrib.contenttypes.models import ContentType
+from django.db import models
+from ulid2 import generate_ulid_as_uuid
+
+from thunderstore.core.mixins import TimestampMixin
+
+
+class Comment(TimestampMixin, models.Model):
+    thread = GenericForeignKey("thread_content_type", "thread_object_id")
```

Comment:
Another thing to consider: Should we somehow worry about cleaning up if the thread object is deleted? Generic relations don't support `on_delete`, which makes it a bit difficult

Reply:
If we're avoiding signals, overwriting `delete()` might be the best way. This gets more complicated as both communities and packages have cascade deletes for `PackageListing`. A clean up cronjob could too I think? That would also make deleting a large community or package more stable. For example, Discord doesn't cascade message deletes when deleting a channel (or even clean them up).

Reply:
A cleanup task is the best solution I'm aware of currently. Ideally any solution to the cleanup should be defined on the `Comment` side of logic rather than what is used as the `thread`, since implementing it on the thread-side will not scale well and is much more prone to mistakes

Reply:
> Ideally any solution to the cleanup should be defined on the `Comment` side of logic rather than what is used as the `thread`

What do you mean?

Reply:
`thread` can be any model, which means if we want to do comment cleanup on thread deletion, it will have to be done on the thread model's `.delete()` function -> we now have to implement it to all models or alternatively somehow restrict which mdoels can be used as a thread.